classes/page_group.php

```diff
@@ -31,6 +31,35 @@ class page_group extends persistent {
     /** The name of the database table. */
     public const TABLE = 'tool_excimer_page_groups';
 
+    /** Name of the cache used in this class. */
+    public const CACHE_NAME = 'page_group_metadata';
+
+    /**
+     * Creates a page_group, either from the cache, or fresh.
+     *
+     * @param string $name
+     * @param string $month
+     * @return page_group
+     */
+    public static function get_page_group(string $name, string $month): page_group {
+        $keydata = ['name' => $name, 'month' => $month];
+        $cachekey = serialize($keydata);
+        $cache = \cache::make('tool_excimer', self::CACHE_NAME);
+        $record = $cache->get($cachekey);
```

Comment:
Could we add a comment to this block, perhaps something related to it attempting to pull out the data from cache first.